pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -428,6 +432,84 @@ func (c ClusterSpec) IsClusterBackupEnabled() bool {
 		c.BackupConfig.BackupStorageLocation.Name != ""
 }
 
+type AuthorizationConfig struct {
+	// Optional: List of enabled Authorization modes (by default 'Node,RBAC')
+	// Important: order matters
+	EnabledModes []string `json:"enabledModes,omitempty"`
+	// Contains the settions for the AuthorizationWebhook if EnabledModes contains Webhook
+	AuthorizationWebhookConfiguration *AuthorizationWebhookConfiguration `json:"authorizationWebhookConfiguration,omitempty"`
+
+	AuthorizationConfigurationFile *AuthorizationConfigurationFile `json:"authorizationConfigurationFile,omitempty"`
+}
+type AuthorizationWebhookConfiguration struct {
+	// The secret containing the webhook configuration
+	SecretName string `json:"secretName"`
+	// The secret Key inside the secret
+	SecretKey string `json:"secretKey"`
+	// the Webhook Version, by default "v1"
+	WebhookVersion string `json:"webhookVersion"`
+}
+
+type AuthorizationConfigurationFile struct {
+	// The secret containing the authorizaion configuration
+	SecretName string `json:"secretName"`
+	// The secret Key containing the AuthorizationConfig k8s object
+	SecretKey string `json:"secretKey"`
+	// the path were the secret should be mounted, by default '/etc/kubernetes/authorization-configs'
+	SecretMountPath string `json:"secretMountPath"`
+}
+
+func (c ClusterSpec) GetAuthorizationModesString() string {
+	if c.AuthorizationConfig != nil && c.AuthorizationConfig.EnabledModes != nil {
+		return strings.Join(c.AuthorizationConfig.EnabledModes, ",")
+	}
+
+	return "Node,RBAC"
+}
```

Comment:
I would also not make this part of the KKP API, but keep it in the apiserver's package.